kernel/src/sched.rs

```diff
@@ -331,19 +349,32 @@ impl Kernel {
                                     allow_address,
                                     allow_size,
                                 }) => {
-                                    let res = platform.with_driver(driver_number, |driver| {
-                                        match driver {
-                                            Some(d) => {
-                                                match process.allow(allow_address, allow_size) {
-                                                    Ok(oslice) => {
-                                                        d.allow(appid, subdriver_number, oslice)
+                                    let permissions: u64 = if let Some(header) =
+                                        tbfheader::parse_and_validate_tbf_header(
```

Comment:
This seems like it should be done in `process.rs` when the process is loaded for the first time, and cached in the `Process` struct, so you don't need to parse and validate on every system call.

Reply:
Good call -- I moved it to the ProcessType trait -- I think this is a much better place for it.